src/main/scala/Timer.scala

```diff
@@ -7,13 +7,33 @@ class Timer extends Module {
   })
   val mtime = RegInit(0.U(64.W))
   val mtimecmp = RegInit("hffffffffffffffff".U(64.W))
-  io.interruptPending := mtime >= mtimecmp
-  mtime := mtime + 1.U
+
+  // Timer Usage:
+  // 
+  // [0]: timer enable
+  // [1]: timer Int enable
+  // [2]: timer Int pending, write 1 to clear it
+  // [3]: 64 bit mode
+  val mtimectrl = RegInit(0.U(32.W))
+
+  io.interruptPending := mtimectrl(2) & mtimectrl(1)
```

Comment:
The spec says:

>  The interrupt remains posted until it is cleared by writing the `mtimecmp` register

I think we should not use `mtimectl` to control the interrupt.

If you want to mask out the intrrupt, you can set `MTIE`.